### PR TITLE
[Experiment] Maven w/ transport-http

### PR DESCRIPTION
**Pull request by cstamas**
This PR creates maven distro (kept up to date with master) that contains latest (locally built) maven-resolver along w/ transport-http and transport-wagon. Also, the build banishes wagon from Maven, but it is still present in lib of distro as maven-compat still uses it.

Try this build locally, to build it:
* build/install locally maven-resolver master
* build locally distro using this PR
* deploy locally the built maven

To test it:
* try to locally build master (due resolver SNAPSHOT dependency) of maven using vanilla distro and this distro (with or without local MRM). To notice the speed difference between two transports, always use empty local repo (to make Maven pull all).
* try out the "provided" checksum features (provided checksums are in this PR, but to build this PR you must set maven dependency to latest released resolver). Using `.mvn/checksums-bad` will fail the build, while `.mvn/checksums-good` will make build pass when plexus-utils:3.3.0 is being fetched.

(Don't forget to disable RAT if you use in-dir local repo)

Example of "provided" use:
```
~/bin/maven/apache-maven-4.0.0-alpha-1-SNAPSHOT-transport-http/bin/mvn clean install  -Daether.artifactResolver.providedChecksumsSource.file.baseDir=/PATH/TO/maven/.mvn/checksums-bad -DlocalRepository=localRepository -Drat.skip=true
```

To switch transport to wagon (default is transport-http) add this parameter:
```
-Daether.priority.WagonTransporterFactory=100
```

### CODE REVIEW

**Comment by cstamas** [comment]
All the failing ITs are actually Maven 2.1 ITs as summarized in this JIRA https://issues.apache.org/jira/browse/MNG-7363

**Comment by cstamas** [comment]
Experiment No2: when using checksums-good build passes as expected, but when using checksums-bad, it will fail like this

```
[INFO] BUILD FAILURE
[INFO] --------------------------------------------------------------------------------------------------------------------------
[INFO] Total time:  54.079 s
[INFO] Finished at: 2021-12-16T17:44:01+01:00
[INFO] --------------------------------------------------------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.0.0:enforce (enforce-maven-version) on project maven-bom: Execution enforce-maven-version of goal org.apache.maven.plugins:maven-enforcer-plugin:3.0.0:enforce failed: Plugin org.apache.maven.plugins:maven-enforcer-plugin:3.0.0 or one of its dependencies could not be resolved: Could not transfer artifact org.codehaus.plexus:plexus-utils:jar:3.3.0 from/to central (https://repo.maven.apache.org/maven2): Checksum validation failed, expected BAD but is cf43b5391de623b36fe066a21127baef82c64022 -> [Help 1]
```

**Review by michael-o — CHANGES_REQUESTED** [review]

File: maven-compat/pom.xml [hunk 1]
```diff
@@ -33,6 +33,10 @@ under the License.
   <name>Maven Compat</name>
   <description>Maven2 classes maintained as compatibility layer.</description>
 
+  <properties>
+    <wagonVersion>3.5.1</wagonVersion>
```
[the remark this answers is not in the document]
Comment (cstamas) [reply maven-compat/pom.xml:37]: ? there is no such thing: https://repo.maven.apache.org/maven2/org/apache/maven/wagon/wagon-http/

[the remark this answers is not in the document]
Comment (michael-o) [reply maven-compat/pom.xml:37]: Right. Screw that. I guess I had another project in mind.

File: pom.xml [hunk 2]
```diff
@@ -575,6 +562,7 @@ under the License.
           <configuration>
             <excludes>
               <exclude>.asf.yaml</exclude>
+              <exclude>.mvn/</exclude>
```
Comment (eolivelli) [inline pom.xml:565]: Nit: this is probably unrelated

Comment (cstamas) [reply pom.xml:565]: Um, no, as part of No2 experiment w/ provided checksums, this PR contains `.mvn/checksums` that makes RAT freak out :smile:

Comment (eolivelli) [reply pom.xml:565]: I see.
Makes sense

**Comment by cstamas** [comment]
Superseded by https://github.com/apache/maven/pull/711

**Pull request closed without merging**